sdks/python/gen_xlang_wrappers.py

```diff
@@ -0,0 +1,420 @@
+  python_wrapper_template = env.get_template("python_xlang_wrapper.template")
+
+  # maintain a list of wrappers to write in each file. if modified destinations
+  # are used, we may end up with multiple wrappers in one file.
+  destinations: Dict[str, List[str]] = {}
+
+  with open(config_file) as f:
+    transforms = yaml.safe_load(f)
+    for config in transforms:
+      default_service = config['default_service']
+      description = config['description']
+      destination = config['destinations']['python']
+      name = config['name']
+      fields = config['fields']
+      identifier = config['identifier']
+
+      parameters = []
+      for param, info in fields.items():
+        pythonic_name = camel_case_to_snake_case(param)
+        param_details = {
+            "name": pythonic_name,
+            "type": info['type'],
+            "description": info['description'],
+        }
+
+        if info['nullable']:
+          param_details["default"] = None
+        parameters.append(param_details)
+
+      # Python syntax requires function definitions to have
+      # non-default parameters first
+      parameters = sorted(parameters, key=lambda p: 'default' in p)
+      default_service = f"BeamJarExpansionService(\"{default_service}\")"
+
+      python_wrapper_class = python_wrapper_template.render(
+          class_name=name,
+          identifier=identifier,
+          parameters=parameters,
+          description=description,
+          default_expansion_service=default_service)
+
+      if destination not in destinations:
+        destinations[destination] = []
+      destinations[destination].append(python_wrapper_class)
+
+  return destinations
+
+
+def write_wrappers_to_destinations(grouped_wrappers: Dict[str, List[str]]):
+  """
+  Takes a dictionary of generated wrapper code, grouped by destination.
+  For each destination, create a new file containing the respective wrapper
+  classes. Each file includes the Apache License header and relevant imports.
+  Note: the Jinja template should already follow linting and formatting rules.
+  """
+  written_files = []
+  for dest, wrappers in grouped_wrappers.items():
+    dest += PYTHON_SUFFIX
+    absolute_dest = os.path.join(PYTHON_SDK_ROOT, *dest.split('/'))
+    with open(absolute_dest, "w") as file:
+      file.write(LICENSE_HEADER.lstrip())
+      file.write(
+          MARKER + "# and should not be edited by hand.\n"
+          "# Refer to the utility at gen_xlang_wrappers.py for more info.\n\n")
+      file.write("# pylint:disable=line-too-long\n\n")
+      file.write(
+          "from apache_beam.transforms.external import "
+          "BeamJarExpansionService\n"
+          "from apache_beam.transforms.external_schematransform_provider "
+          "import ExternalSchemaTransform\n")
+      for wrapper in wrappers:
+        file.write(wrapper + "\n")
+    written_files.append(absolute_dest)
+
+  # We only make a best effort attempt to format with yapf because not all
```

Comment:
Would it be easier to simply make yapf a requirement?

Reply:
Yup that works!